lib/pbench/agent/results.py

```diff
@@ -149,23 +149,26 @@ class CopyResultTb:
     """CopyResultTb - Use the server's HTTP PUT method to upload a tarball
     """
 
-    chunk_size = 4096
+    CHUNK_SIZE = 65536
 
     def __init__(
         self, controller: str, tarball: str, config: PbenchAgentConfig, logger: Logger
     ):
+        self.controller = controller
```

Comment:
Do we need to protect the character set for header values? Seems we're basically allowed to use any ASCII octets, including LWS (linear whitespace), so full HTTP encoding would be inappropriate; but a scan for non-ASCII characters in the controller name wouldn't be unreasonable. (Especially as we transition from "filename" to "metadata".)

Reply:
We should be covered if we first `validate_hostname(controller)`.

Reply:
Just to confirm, "as we transition from 'filename' to 'metadata'", are we sure we want to constrain ourselves to valid hostnames for "controller" values?  (We want to outlaw colons, for instance?)

Reply:
We already validate host names for `pbench-move-results`.

We use `validate_hostname()` here on the agent side, just before sending things off to the server, to check that we are not going to send the server a bad value (see the upcoming commit).  This gives the user feedback sooner than if it fails on the server side.

As you can imagine, the server needs to validate API parameters regardless.  In other words, this change does not in any way relieve the server of its responsibility of validating API parameters.